crates/indexer/src/helper.rs

```diff
@@ -0,0 +1,191 @@
+
+    let mut u128_bytes = [0u8; 16];
+    u128_bytes.copy_from_slice(&data[41..57]);
+    let amount = u128::from_le_bytes(u128_bytes);
+    (from_address, to_address, amount)
+}
+
+pub fn parse_log(item: &LogItem) -> Result<GwLog> {
+    let service_flag: u8 = item.service_flag().into();
+    let raw_data = item.data().raw_data();
+    let data = raw_data.as_ref();
+    match service_flag {
+        GW_LOG_SUDT_TRANSFER => {
+            let sudt_id: u32 = item.account_id().unpack();
+            if data.len() != (1 + 20 + 20 + 16) {
+                return Err(anyhow!("Invalid data length: {}", data.len()));
+            }
+            let (from_address, to_address, amount) = parse_sudt_log_data(data);
+            Ok(GwLog::SudtTransfer {
+                sudt_id,
+                from_address,
+                to_address,
+                amount,
+            })
+        }
+        GW_LOG_SUDT_PAY_FEE => {
+            let sudt_id: u32 = item.account_id().unpack();
+            if data.len() != (1 + 20 + 20 + 16) {
+                return Err(anyhow!("Invalid data length: {}", data.len()));
+            }
+            let (from_address, block_producer_address, amount) = parse_sudt_log_data(data);
+            Ok(GwLog::SudtPayFee {
+                sudt_id,
+                from_address,
+                block_producer_address,
+                amount,
+            })
+        }
+        GW_LOG_POLYJUICE_SYSTEM => {
+            if data.len() != (8 + 8 + 20 + 4) {
+                return Err(anyhow!(
+                    "invalid system log raw data length: {}",
+                    data.len()
+                ));
+            }
+
+            let mut u64_bytes = [0u8; 8];
+            u64_bytes.copy_from_slice(&data[0..8]);
+            let gas_used = u64::from_le_bytes(u64_bytes);
+            u64_bytes.copy_from_slice(&data[8..16]);
+            let cumulative_gas_used = u64::from_le_bytes(u64_bytes);
+
+            let created_address = {
+                let mut buf = [0u8; 20];
+                buf.copy_from_slice(&data[16..36]);
+                buf
+            };
+            let mut u32_bytes = [0u8; 4];
+            u32_bytes.copy_from_slice(&data[36..40]);
+            let status_code = u32::from_le_bytes(u32_bytes);
+            Ok(GwLog::PolyjuiceSystem {
+                gas_used,
+                cumulative_gas_used,
+                created_address,
+                status_code,
+            })
+        }
+        GW_LOG_POLYJUICE_USER => {
+            let mut offset: usize = 0;
+            let mut address = [0u8; 20];
+            address.copy_from_slice(&data[offset..offset + 20]);
+            offset += 20;
+            let mut data_size_bytes = [0u8; 4];
+            data_size_bytes.copy_from_slice(&data[offset..offset + 4]);
+            offset += 4;
+            let data_size: u32 = u32::from_le_bytes(data_size_bytes);
+            let mut log_data = vec![0u8; data_size as usize];
```

Comment:
Potential memory attacks, please check data_size before allocating a vector

Reply:
@Flouse Do we have max size of this data?

Reply:
According to [Ethereum Yellow Paper (Page 5)](https://ethereum.github.io/yellowpaper/paper.pdf), there is no size limit to the log_data.

But, as you know, Polyjuice is running on CKB_VM.
Only about 1MB heap memory can be [malloced](https://github.com/nervosnetwork/godwoken-polyjuice/blob/main/c/polyjuice.h#L636) in CKB_VM.

Reply:
I think a reasonable value is 4K or 8K, we can revisit this value if data_size is longer than this.

Reply:
Will throw an error if data size greater than 4KB.